### PR TITLE
fix: fromDecimal Auto delimiter now correctly parses multiple numbers

### LINKED ISSUE #2217: Bug report: fromDecimal(...) function does not correctly handle delim parameter passed as default ("Auto")
**Describe the bug**
When fromDecimal is called without specifying a value for the "delim" parameter, or when delim is passed as "Auto" it will only parse the first number in the passed text ignoring any further numbers whatever separator has been used.

**To Reproduce**
Steps to reproduce the behaviour or a link to the recipe / input used to cause the bug:

See bug report #2139 which arises as a consequence of this issue.

**Expected behaviour**
delim = "Auto" case should be handled correctly, in a manner similar to the fromHex(...) function.

**Additional context**
fromHex could be used as a guide towards a correct implementation, noting that there are other differences because pairs of hex digits can be packed exactly into bytes whereas for decimal up to 3 digits can be required.

### PR DESCRIPTION
**Pull request by min23asdw**
## Issue

Fixes #2217
Ref: #2221 (closed due to unsigned CLA)

## Description

`fromDecimal()` did not correctly handle the `delim` parameter when set to `"Auto"`.

**Root cause:** `Utils.charRep("Auto")` returns `undefined` (no "Auto" entry in the mapping), so `data.split(undefined)` returns the entire string as a single array element — only the first number gets parsed.

**Fix:** When `delim === "Auto"`, split on regex `/[^\d-]+/` (any non-digit, non-minus characters), consistent with how `fromHex()` handles its Auto delimiter. Existing delimiter behavior is unchanged.

## Changes

- `src/core/lib/Decimal.mjs` — Auto delimiter uses regex split instead of `Utils.charRep`
- `tests/operations/tests/FromDecimal.mjs` — Added 5 test cases for Auto delimiter (space, comma, mixed, newline, signed values)

## Before / After

| Input | Before | After |
|-------|--------|-------|
| `"72 101 108 108 111"` (Auto) | `[72]` → `"H"` | `[72,101,108,108,111]` → `"Hello"` |
| `"72,101,108,108,111"` (Auto) | `[72]` → `"H"` | `[72,101,108,108,111]` → `"Hello"` |
| `"72, 101 : 108; 108\t111"` (Auto) | `[72]` → `"H"` | `[72,101,108,108,111]` → `"Hello"` |

## Verification

- `npx grunt lint` — all tasks pass
- All existing and new test cases pass

## Note on edge cases

Two edge cases with malformed input were tested — neither is a security concern:

| Input | Output | Reason |
|-------|--------|--------|
| `"--1"` | `[NaN]` | Double minus is not a valid number — `parseInt("--1")` returns `NaN`. This is consistent with garbage-in, garbage-out behavior. |
| `"72;alert(1);101"` | `[72, 1, 101]` | The regex treats `alert(` as a delimiter, so the `1` inside the parentheses is parsed as a number. No code execution occurs — `fromDecimal` only produces an integer array via `parseInt`. This matches `fromHex` Auto behavior. |

 
  ▎ This pull request was created with the assistance of Claude  and Antigravity, to suggest ideas on where it might be broken and to help write the tests.  then searched online to verify whether those suggestions were correct. 
I have reviewed and tested all code changes. All changes were manually verified.

### CODE REVIEW

**Review by GCHQDeveloper581 — CHANGES_REQUESTED** [review]
We're currently having some internal discussions on how/whether we accept AI created/assisted submissions, so there may be a delay in responding further if/once my other comment is addressed.

**Comment by min23asdw** [comment]
@GCHQDeveloper581   would you mind if it shold have  'FROM_DECIMAL_DELIM_OPTIONS'
https://github.com/gchq/CyberChef/pull/2270/changes#diff-758aa870501726a364e22480816d2b39af0509502254f69a8258c25aca760b17R44

becase if it don't have https://github.com/gchq/CyberChef/pull/2270#issuecomment-4109138679 'auto' mode it will can't not access by UI,


cc https://github.com/gchq/CyberChef/pull/2308

File: src/core/lib/Decimal.mjs [hunk 1]
```diff
@@ -24,14 +25,20 @@ import Utils from "../Utils.mjs";
...
         output[i] = parseInt(byteStr[i], 10);
     }
     return output;
 }
+
+
+/**
+ * From Decimal delimiters.
+ */
+export const FROM_DECIMAL_DELIM_OPTIONS = [...DELIM_OPTIONS, "Auto"];
```
Comment (GCHQDeveloper581) [inline src/core/lib/Decimal.mjs]: I think this is better placed near the top of src/core/operations/FromDecimal.mjs

**Review by GCHQDeveloper581 — COMMENTED** [review]
One minor tweak, but otherwise looks good.

File: src/core/lib/Decimal.mjs (hunk 1 above)
Comment (min23asdw) [reply src/core/lib/Decimal.mjs]: I see..  It better and it make it can still use `DELIM_OPTIONS`  in src\core\operations\FromDecimal.mjs

**Review by GCHQDeveloper581 — APPROVED** [review]
Looks good!

Thanks for the contribution, and apologies for the time it has taken to process it.
